R/simulate_data.R

```diff
@@ -0,0 +1,611 @@
+#' drop-out and data from this visit and all subsequent visits are consequently set to missing.
+#'   In addition, in case the subject is still on treatment at the subject's (first)
+#' visit affected by drop-out (i.e. if this occurs prior to ICE1),
+#'   then dropout also triggers discontinuation of study drug and a corresponding ICE2
+#' (study treatment discontinuation due to NSDCR reasons) is generated.
+#' - Adjust trajectories after ICE according to the given assumption expressed with
+#' the `post_ice_traj` argument.
+#' - Simulate additional intermittent missing outcome data as per arguments `pars_c$prob_miss`
+#' and `pars_t$prob_miss`.
+#'
+#' The probability of the ICE after each visit is modeled according to the following
+#' logistic regression model:
+#' `~ 1 + I(visit == 0) + ... + I(visit == n_visits-1) + I((x-alpha))` where:
+#' - `n_visits` is the number of visits (including baseline).
+#' - `alpha` is the baseline outcome mean.
+#' The term `I((x-alpha))` specifies the dependency of the probability of the ICE on
+#' the current outcome value.
+#' The corresponding regression coefficients of the logistic model are defined as follows:
+#' The intercept is set to 0, the coefficients corresponding to discontinuation after
+#' each visit for a subject with outcome equal to
+#' the mean at baseline are set according to parameters `pars_c$prob_ice1` (`pars_t$prob_ice1`),
+#' and the regression coefficient associated with the covariate `I((x-alpha))` is set
+#' to `log(pars_c$or_outcome_ice)` (`log(pars_t$or_outcome_ice)`).
+#'
+#' Please note that the baseline outcome cannot be missing nor be affected by any ICEs.
+#'
+#' @returns A `data.frame` containing the simulated data. It includes the following variables:
+#' - `id`: Factor variable that specifies the id of each subject.
+#' - `visit`: Factor variable that specifies the visit of each assessment. Visit `0` denotes
+#' the baseline visit.
+#' - `group`: Factor variable that specifies which treatment group each subject belongs to.
+#' - `outcome_bl`: Numeric variable that specifies the baseline outcome.
+#' - `outcome_noICE`: Numeric variable that specifies the longitudinal outcome assuming
+#' no ICEs.
+#' - `ind_ice1`: Binary variable that takes value `1` if the corresponding visit is
+#' affected by ICE1 and `0` otherwise.
+#' - `dropout_ice1`: Binary variable that takes value `1` if the corresponding visit is
+#' affected by the drop-out following ICE1 and `0` otherwise.
+#' - `dropout2`: Binary variable that takes value `1` if the corresponding visit is affected
+#' by the uninformative study drop-out.
+#' - `ind_ice2`: Binary variable that takes value `1` if the corresponding visit is affected
+#' by ICE2.
+#' - `outcome`: Numeric variable that specifies the longitudinal outcome including ICE1, ICE2
+#' and the intermittent missing values.
+#'
+#' @export
+simulate_data <- function(pars_c, pars_t, post_ice_traj, strategies = getStrategies()) {
```

Comment:
Do you think it would be worth setting some default values for pars_c and pars_t so that users can just run the function `simulate_data()` if they just want some simple example data without having to put any thought into it ? (i.e. to just mess around)

Reply:
At the moment, the easiest is that the user sets only mean trajectories and covariance matrices (using `set_simul_pars()`), and what happens is that data are simulated without any ICE. Is that simple enough? I think every longitudinal data simulation requires at least the specification of mean and covariances.

Reply:
I'm more thinking of people just wanting to run the functions to see how it works, I think for them it would be nice to be able to just run `simulate_data()` and have some data available even if that data is meaningless. I.e. just a quick easy way to use the primary rbmi functions without having to configure anything ?

Reply:
Couldn't we use `simulate_test_data()` for this purpose? I think it is currently set-up to work without any argument, isn't it? I am not sure I would like to set defaults for mean and covariances here.. How important is in your opinion to have this feature in `simulate_data()`?

Reply:
@nociale , whats your reason for not wanting to set defaults in `simulate_data()` I'm thinking ideally we would not expose `simulate_test_data()` to users as to avoid any confusion.

Reply:
@gowerc We need to decide whether to provide some defaults so that one can run `simulate_data()` without any argument. Currently one has to set at least number of patients, mean trajectories and covariance matrices, and `post_ice1_traj`. My idea is that if the user specifies parameters only in one arm, then if he just uses the default for the other arm he could obtain some non-sense data. I am happy to provide defaults to make it run easily, but we should be aware that non-sense things are a bit less controllable. One "in-between" solution could also be the following:
- the user needs to specify one argument, `pars_c` (so at least mean, covariance matrix and number of subjects in the control arm)
- by default, `pars_t = pars_c`
- by default, post_ice1_traj = "MAR"
This would simulate trajectories assuming no treatment effect.

I agree that can avoid exposing `simulate_test_data()`.